tests/integration/test_integration.py

```diff
@@ -1,8 +1,11 @@
+from __future__ import annotations
```

Comment:
Do we need this?

Reply:
Yes, since `QueryCollector` references itself